libraries/Microsoft.Bot.Builder.Azure/AzureBlobStorage.cs

```diff
@@ -0,0 +1,232 @@
+        {
+            _storageAccount = storageAccount ?? throw new ArgumentNullException(nameof(storageAccount));
+            _containerName = containerName ?? throw new ArgumentNullException(nameof(containerName));
+            
+            // Checks if a container name is valid
+            NameValidator.ValidateContainerName(containerName);
+        }
+
+        /// <summary>
+        /// Deletes entity blobs from the configured container
+        /// </summary>
+        /// <param name="keys">An array of entity keys</param>
+        /// <returns></returns>
+        public async Task DeleteAsync(string[] keys, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            if (keys == null) throw new ArgumentNullException(nameof(keys));
+
+            var blobContainer = await GetBlobContainer().ConfigureAwait(false);
+            await Task.WhenAll(
+                keys.Select(key =>
+                {
+                    var blobName = GetBlobName(key);
+                    var blobReference = blobContainer.GetBlobReference(blobName);
+                    return blobReference.DeleteIfExistsAsync();
+                }));
+        }
+
+        /// <summary>
+        /// Retrieve entities from the configured blob container
+        /// </summary>
+        /// <param name="keys">An array of entity keys</param>
+        /// <returns></returns>
+        public async Task<IDictionary<string, object>> ReadAsync(string[] keys, CancellationToken cancelationToken = default(CancellationToken))
+        {
+            if (keys == null) throw new ArgumentNullException(nameof(keys));
+
+            var blobContainer = await GetBlobContainer().ConfigureAwait(false);
+
+            var readTasks = new List<Task<KeyValuePair<string, object>>>();
+
+            foreach(var key in keys)
+            {
+                readTasks.Add(ReadIndividualKey(key));
+            }
+
+            await Task.WhenAll(readTasks);
+
+            // Project back the entries that were read, filtering out any entries that were not found.
+            // This gives us a Dictionary(key, value) 
+            var items = readTasks.Select(readTask => readTask.Result)
+                .Where(kvp => kvp.Key != null)
+                .ToDictionary(kvp => kvp.Key, kvp => kvp.Value);
+          
+            return items; 
+
+            async Task<KeyValuePair<string, object>> ReadIndividualKey(string key)
+            {
+                var blobName = GetBlobName(key);
+                var blobReference = blobContainer.GetBlobReference(blobName);
+
+                try
+                {
+                    using (var blobStream = await blobReference.OpenReadAsync())
+                    using (var jsonReader = new JsonTextReader(new StreamReader(blobStream)))
+                    {
+                        var obj = JsonSerializer.Deserialize(jsonReader);
+
+                        if (obj is IStoreItem storeItem)
```

Comment:
Semi-related since you're bringing this logic back: I thought we all had agreed we were going to get rid of `IStoreItem` and, more specifically, `ETag`???

Reply:
I think we do expect to get rid of that, and this is just an oversight on Tom's part.